Improper input validation on -o flag

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->

**What happened**:
I want to write my docker-registry secret to a file.  This command line option is supported, so I run:

```bash
kubectl create secret docker-registry example-token --dry-run=client <credential flags ommitted> -o  example-token.yaml
```

This command results in a segmentation fault, as the correct usage of the command line option is `-o yaml > example-token.yaml`.  Error output is as follows:

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1918de0]
```

**What you expected to happen**:
I expect the command in this case to display a specific command regarding usage of the `-o` flag rather than failing with a segmentation fault. 

**How to reproduce it (as minimally and precisely as possible)**:
<!-- Please make sure you are able to reproduce the bug using a supported version and version skew of Kubernetes. See https://kubernetes.io/docs/setup/release/version-skew-policy for which versions and are currently supported.
-->
```bash
kubectl create secret docker-registry example-token --dry-run=client  --docker-server=<hidden> --docker-username=<hidden> --docker-password=<hidden> --docker-email=<hidden> -o example-token.yaml
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes client and server versions (use `kubectl version`): 1.22
- Cloud provider or hardware configuration: Azure
- OS (e.g: `cat /etc/os-release`): Ubuntu 20.04

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_docker.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_docker.go
@@ -198,13 +198,13 @@ func (o *CreateSecretDockerRegistryOptions) Complete(f cmdutil.Factory, cmd *cob
 
 	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
-		return nil
+		return err
 	}
 
 	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
 	printer, err := o.PrintFlags.ToPrinter()
 	if err != nil {
-		return nil
+		return err
 	}
 
 	o.PrintObj = func(obj runtime.Object) error {

